### PR TITLE
Implemented Attribute Certificate Authority.

### PR DESCRIPTION
**Pull request by andresgaragiola**
This pull request include the ACA (Attribute Certificate Authority) implementation.

### CODE REVIEW

**Comment by andresgaragiola** [comment]
Signed-off-by: Andres Garagiola garagiol@ar.ibm.com/andresgaragiola@gmail.com

**Comment by srderson** [comment]
@andresgaragiola Please rebase on master as there are merge conflicts. Thanks

**Comment by andresgaragiola** [comment]
@srderson I have fixed the conflicts and committed the code again.

**Comment by srderson** [comment]
Is the "Attribute Certificate Authority" documented somewhere? I could not find existing documentation and did not see anything in the PR. Is there an associated issue? Or some test cases I could follow? I'm having trouble reviewing this PR without context or background. Is there a reviewer you could suggest? @smithbk or @binhn do you know about this change?

**Comment by andresgaragiola** [comment]
@srderson this PR is related with #1139 about Attribute Based Access Control. Attribute Certificate Authority is a new feature within the membership services that enables prove the authenticity of an attribute associated to a specific user.
- During user enrollment the ECA invokes a fetch request to the ACA in order to populate in the ACA database the external attributes (currently this attributes are defined in the membersrvc.yml file). 
- During TCert generation the TCA invokes a register request to the ACA in order to determine which attributes specified by the user are authentic and which are not. The generated TCert only includes valid attributes according to the user ownership and the validity period.

We have been working in this functionality with @dulcep , @diegomasini and @raullaprida. I'm documenting it in a wiki page.

**Comment by srderson** [comment]
Please post a link to the wiki page once available. Thanks.

**Comment by andresgaragiola** [comment]
@srderson  [here](https://github.com/hyperledger/fabric/wiki/Attribute-Certificate-Authority) is the link to the ACA documentation. Any doubt let me know. Thanks

File: membersrvc/ca/eca.go [hunk 1]
```diff
@@ -200,6 +210,10 @@ func (eca *ECA) startECAA(srv *grpc.Server) {
 	pb.RegisterECAAServer(srv, &ECAA{eca})
 }
 
+func (eca *ECA) ivokeACAFetchAttributes(id, affiliation string ) { 
+	//eca.aca.
```
Comment (BiscuitBomber) [inline membersrvc/ca/eca.go:214]: ?

File: membersrvc/ca/tca.go [hunk 2]
```diff
@@ -42,14 +45,17 @@ import (
...
+
 )
 
 var (
 	// TCertEncTCertIndex is the ASN1 object identifier of the TCert index.
 	TCertEncTCertIndex = asn1.ObjectIdentifier{1, 2, 3, 4, 5, 6, 7}
 
 	// TCertEncEnrollmentID is the ASN1 object identifier of the enrollment id.
-	TCertEncEnrollmentID = asn1.ObjectIdentifier{1, 2, 3, 4, 5, 6, 8}
+	TCertEncEnrollmentID = asn1.ObjectIdentifier{1, 2, 3, 4, 5, 6, 8}
```
Comment (BiscuitBomber) [inline membersrvc/ca/tca.go:58]: ?

File: docs/aca.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        Signature signature = 5;                       // Signed using the TCA private key.
+    }
+
+    message ACAAttrResp {
+        enum StatusCode {
+        	FULL_SUCCESSFUL     = 000;
+           PARTIAL_SUCCESSFUL  = 001;
+    	    NO_ATTRIBUTES_FOUND = 010;
+    	    FAILURE	          = 100;
+    	 }
```
Comment (BiscuitBomber) [inline docs/aca.md:91]: What's all this inconsistent white-spacing/indenting business?

File: docs/aca.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+    rpc RequestAttributes(ACAAttrReq) returns (ACAAttrResp);
+
+    message ACAAttrReq {
+        google.protobuf.Timestamp ts = 1;
+        Identity id = 2;
+        Cert eCert = 3;                                // ECert of involved user.
+        repeated TCertAttributeHash attributes = 4;    // Pairs attribute-key, attribute-value-hash
+        Signature signature = 5;                       // Signed using the TCA private key.
+    }
```
Comment (BiscuitBomber) [inline docs/aca.md:83]: Also, lines 78-82 are very different than lines 86-90 attempting to have all the '=' signs aligned.

**Comment by BiscuitBomber** [comment]
Any reason for closing this PR? (And if so, what is it?) Thanks

**Comment by andresgaragiola** [comment]
@BiscuitBomber we are documenting all ABAC functionality previously to send the PR again. I will send a PR contained this feature soon.

**Pull request closed without merging**